### PR TITLE
added feedback controller and message model and add routes for it

### PR DESCRIPTION
**Pull request by nooralnabi**

### CODE REVIEW

File: app/controllers/api/v1/public_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class Api::V1::PublicController < ApplicationController
+
+  def custom_feedback
+    begin
+      @message = PointlessFeedback::Message.new(message_params)
```
Comment (MwMaki) [inline app/controllers/api/v1/public_controller.rb]: no need for the @ sign

File: app/controllers/api/v1/public_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      :name,
+      :topic,
+    ])
+  end
+
+  def prepare_request_data
+    customer = Customer.find(params[:customer_id])
+    {
+      ip_address: request.remote_ip,
+      referer: request.referer,
```
Comment (MwMaki) [inline app/controllers/api/v1/public_controller.rb]: what do you mean by referer: request.referer, ?

Comment (nooralnabi) [reply app/controllers/api/v1/public_controller.rb]: @MwMaki this code in UI pure,  so I think we want the referer for the customer but is not necessary

File: app/controllers/api/v1/public_controller.rb (hunk 1 above)
Comment (nooralnabi) [reply app/controllers/api/v1/public_controller.rb]: ok

File: app/controllers/api/v1/public_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  def custom_feedback
+    begin
+      message = PointlessFeedback::Message.new(message_params)
+      message.request_data = prepare_request_data
+      if message.save
+        render json: {status: I18n.t('views.pointless_feedback.messages.saved')}, status: :ok
+      else
+        render json: {error: message.errors.full_messages}, status: :unprocessable_entity
+      end
+    rescue Exception => e
```
Comment (MwMaki) [inline app/controllers/api/v1/public_controller.rb]: Please remove the begin and rescue everywhere

File: config/environments/development.rb [hunk 4]
```diff
@@ -44,7 +44,18 @@
...
   config.active_record.verbose_query_logs = true
 
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.perform_deliveries = true
 
+
+  config.action_mailer.smtp_settings = {
+  address: 'smtp.gmail.com',
+  port: 587,
+  user_name: ENV['GMAIL_USER'],
```
Comment (MwMaki) [inline config/environments/development.rb]: Please update the application.yml.example with those variables

File: config/initializers/pointless_feedback.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  # Configure the topics for the user to choose from on the feedback form
+  config.message_topics = ['Error on page', 'Website Feedback', 'Bug Report', 'Questions', 'Other']
+
+  # ==> Email Configuration
+  # Configure feedback email properties (disabled by default)
+  # Variables needed for emailing feedback
+  config.email_feedback            = true
+  config.send_from_submitter       = true
+  config.from_email                = ENV['FROM_EMAIL']
+  config.to_emails                 = ENV['TO_EMAILS']
```
Comment (MwMaki) [inline config/initializers/pointless_feedback.rb]: Please update the application.yml.example with those variables

Comment (nooralnabi) [reply config/initializers/pointless_feedback.rb]: ok

File: app/controllers/api/v1/public_controller.rb (hunk 3 above)
Comment (nooralnabi) [reply app/controllers/api/v1/public_controller.rb]: ok

**Pull request merged**